test-data/unit/check-unreachable-code.test

```diff
@@ -841,6 +841,38 @@ def baz(x: int) -> int:
         return x
 [builtins fixtures/exception.pyi]
 
+[case testUnreachableFlagExcepts]
+# flags: --warn-unreachable
+from typing import Type, NoReturn, Tuple
+
+def error() -> NoReturn: ...
+
+ignore: Type[Exception]
+exclude: Tuple[Type[Exception], ...]
+class ContinueOnError: pass
+
+try:
+    error()
+except ignore:
+    pass
+except exclude:
+    pass
+except ContinueOnError:            # E: Exception type must be derived from BaseException
+    pass
+except Exception:
+    pass
+except Exception as err:           # E: Except is unreachable, "Exception" has already been caught
+    pass
+except RuntimeError:               # E: Except is unreachable, superclass "Exception" of "RuntimeError" has already been caught
+    pass
+except (NotImplementedError, ):
```

Comment:
I think that this should be reported as unreachable as well.

Reply:
Do you mean I should support tuples in general or only this special case? 

My feeling is that this already makes the code more complex, because the simple check `expr and isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo)` to support everything that is not supported is no longer sufficient.

Reply:
We can defer this to the next PR I guess 🙂 
But, please don't forget to open an issue for it!